test/integration/cdi/example-driver/app/kubeletplugin.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package app
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	cdiapi "github.com/container-orchestrated-devices/container-device-interface/pkg/cdi"
+	specs "github.com/container-orchestrated-devices/container-device-interface/specs-go"
+	"k8s.io/klog/v2"
+	drapbv1 "k8s.io/kubernetes/pkg/kubelet/apis/dra/v1alpha1"
+	plugin "k8s.io/kubernetes/test/integration/cdi/example-driver/kubeletplugin"
+)
+
+const (
+	deviceName = "exampledevice"
+	cdiVersion = "0.2.0"
+)
+
+type examplePlugin struct {
+	cdiDir                 string
+	driverName             string
+	draAddress             string
+	pluginRegistrationPath string
+}
+
+// newExamplePlugin returns an initialized examplePlugin instance.
+func newExamplePlugin(config examplePlugin) (*examplePlugin, error) {
```

Comment:
so, you're passing examplePlugin struct as a parameter to create examplePlugin struct? This looks quite strange to be honest. Can you just pass 4 parameters instead?

Reply:
sure.

Reply:
There's actually a good argument for passing a struct: it's easy to mix up the order of parameters when a function takes four parameters of the same type (strings in this case). Optional parameters are easy with a struct and impossible with args.

No need to change this again, I just wanted to point it out.